packages/suite-base/src/App.test.tsx

```diff
@@ -0,0 +1,191 @@
+};
+
+describe("App Component", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("renders without crashing", () => {
+    setup();
+    expect(screen.getByTestId("color-scheme-theme")).toBeDefined();
+    expect(screen.getByTestId("css-baseline")).toBeDefined();
+    expect(screen.getByTestId("error-boundary")).toBeDefined();
+    expect(screen.getByTestId("multi-provider")).toBeDefined();
+    expect(screen.getByTestId("document-title-adapter")).toBeDefined();
+    expect(screen.getByTestId("send-notification-toast-adapter")).toBeDefined();
+    expect(screen.getByTestId("panel-catalog-provider")).toBeDefined();
+    expect(screen.getByTestId("workspace-component")).toBeDefined();
+  });
+
+  it("renders GlobalCss when enableGlobalCss is true", () => {
+    setup({ enableGlobalCss: true });
+    expect(screen.getByTestId("global-css")).toBeDefined();
+  });
+
+  it("throw exception when enableGlobalCss is false", () => {
+    setup({ enableGlobalCss: false });
+    expect(() => screen.getByTestId("global-css")).toThrow();
+  });
+
+  it("adds and removes contextmenu event listener on mount/unmount", () => {
+    const addEventListenerSpy = jest.spyOn(document, "addEventListener");
+    const removeEventListenerSpy = jest.spyOn(document, "removeEventListener");
+
+    const { unmount } = setup();
+
+    expect(addEventListenerSpy).toHaveBeenCalledWith("contextmenu", expect.any(Function));
+    expect(removeEventListenerSpy).not.toHaveBeenCalled();
+    unmount();
+    expect(removeEventListenerSpy).toHaveBeenCalledWith("contextmenu", expect.any(Function));
+  });
+
+  it("renders LaunchPreference component when enableLaunchPreferenceScreen is true", () => {
+    setup({ enableLaunchPreferenceScreen: true });
+    expect(screen.getByTestId("launch-preference")).toBeDefined();
+  });
+
+  it("does not render LaunchPreference component when enableLaunchPreferenceScreen is false", () => {
+    setup({ enableLaunchPreferenceScreen: false });
+    expect(screen.queryByTestId("launch-preference")).toBeNull();
+  });
+
+  it("passes deepLinks and onAppBarDoubleClick to Workspace", () => {
+    const mockDeepLinks = ["link1", "link2"];
+    const mockOnAppBarDoubleClick = jest.fn();
+    expect(Workspace).not.toHaveBeenCalled();
+
+    setup({
+      deepLinks: mockDeepLinks,
+      onAppBarDoubleClick: mockOnAppBarDoubleClick,
+    });
+
+    // Ensure Workspace receives the correct props by spying on Workspace
+    expect(Workspace).toHaveBeenCalledWith(
+      {
+        deepLinks: mockDeepLinks,
+        onAppBarDoubleClick: mockOnAppBarDoubleClick,
+      },
+      {},
+    );
+  });
+});
+
+describe("App Component MultiProvider Tests", () => {
+  const expectedProviders = [
+    StudioToastProvider,
+    StudioLogsSettingsProvider,
+    ProblemsContextProvider,
+    CurrentLayoutProvider,
+    UserProfileLocalStorageProvider,
+    LayoutManagerProvider,
+    LayoutStorageContext.Provider,
```

Comment:
Here it should have 13 providers, because it is 13 providers that is sent to Multiprovider component every single time, independent of flags or ifs:

1. TimelineInteractionStateProvider
2. UserScriptStateProvider
3. ExtensionMarketplaceProvider
4. ExtensionCatalogProvider
5. PlayerManager
6. EventsProvider
7. StudioToastProvider
8. StudioLogsSettingsProvider
9. ProblemsContextProvider
10. CurrentLayoutProvider
11. UserProfileLocalStorageProvider
12. LayoutManagerProvider
13. LayoutStorageContext.Provider